tiatoolbox/models/dataset/dataset_abc.py

```diff
@@ -81,10 +95,13 @@ def _check_input_integrity(self: PatchDatasetABC, mode: str) -> None:
                 shapes = [self.load_img(v).shape for v in self.inputs]
                 self.data_is_npy_alike = False
 
-            if is_all_npy:
+            elif self._are_npy_like(self.inputs):
                 shapes = [v.shape for v in self.inputs]
                 self.data_is_npy_alike = True
 
+            else:
+                raise ValueError(msg)
+
             if shapes:
```

Comment:
This if statement is only covered for the True condition not for False. The rest is fine and this PR is ready to merge.

https://app.codecov.io/gh/TissueImageAnalytics/tiatoolbox/pull/726/blob/tiatoolbox/models/dataset/dataset_abc.py